src/components/AnnotationBrowser/AnnotationCSVDialog.vue

```diff
@@ -121,6 +161,24 @@ export default class AnnotationCsvDialog extends Vue {
   @Prop()
   readonly propertyPaths!: string[][];
 
+  propertyExportMode: "all" | "selected" | "listed" = "all";
+  propertyFilter: string = "";
+  selectedPropertyPaths: PropertyPathItem[] = [];
+
+  get filteredPropertyItems() {
+    return this.propertyPaths
+      .filter((path) => {
+        if (!this.propertyFilter) return true;
+        const name = this.propertyStore.getFullNameFromPath(path);
+        return name?.toLowerCase().includes(this.propertyFilter.toLowerCase());
+      })
+      .map((path) => ({
+        name: this.propertyStore.getFullNameFromPath(path) || "",
+        path: path,
+        pathString: path.join("."),
+      }));
```

Comment:
```ts
return (
        this.propertyFilter
          ? this.propertyPaths
          : this.propertyPaths.filter((path) => {
              const name = this.propertyStore.getFullNameFromPath(path);
              return name
                ?.toLowerCase()
                .includes(this.propertyFilter.toLowerCase());
            })
      ).map((path) => ({
        name: this.propertyStore.getFullNameFromPath(path) || "",
        path: path,
        pathString: path.join("."),
      }));
```
This would eliminate the iteration if `propertyFilter` is set

Reply:
Good catch, thanks!